growthbook/growthbook
# List features api

### Features and Changes

This PR is a subset of the following: https://github.com/growthbook/growthbook/pull/499. The only included endpoint is list features. The reasoning for this being is: In the case that there is an error with the new RESTful API it is a lot safer to roll a single endpoint out at a time.

The new endpoint (and future endpoints) omits the following properties from the FeatureInterface: "organization," "drafts," and "revisions." organization is the organization id and is considered sensitive data nor is it required to be sent back. drafts and revisions are both primarily for behind-the-scenes use by GrowthBook and would otherwise bloat the return from the API.

### Testing

1. Start the application with `yarn dev`
2. Start the local docs  `yarn dev` from `/packages/docs` 
3. Follow the instructions found at http://localhost:3200/apidocs/authentication
4. Use a service similar to postman or whatever you prefer to test the endpoint `http://localhost:3100/api/v1/features` make sure to include your `access_token` as a header with the request (documented in the docs).

### Screenshots

The docs have been trimmed to only include access_token requests
![image](https://user-images.githubusercontent.com/36757028/193697464-051fab73-da94-4e78-a59b-57c5d9b4d7c9.png)

The returned features object that does not include the props mentioned above
![image](https://user-images.githubusercontent.com/36757028/193697554-c6ae66bf-248e-47a7-8d83-8aae9a29fee6.png)

![image](https://user-images.githubusercontent.com/36757028/193697598-4533ba4c-8036-4882-9c52-2ea03a72ed78.png)
